### PR TITLE
RavenDB-9813 : test replication of expired and deleted revisions

### PR DESCRIPTION
**Pull request by aviv86**
https://issues.hibernatingrhinos.com/issue/RavenDB-9813

### CODE REVIEW

File: src/Raven.Server/Documents/Revisions/RevisionsStorage.cs [hunk 1]
```diff
@@ -482,9 +482,18 @@ private BlittableJsonReaderObject RecreateCountersIfNeeded(DocumentsOperationCon
...
+            if (configuration.MinimumRevisionsToKeep.HasValue)
+            {
+                numberOfRevisionsToDelete = revisionsCount - configuration.MinimumRevisionsToKeep.Value;
+                if (numberOfRevisionsToDelete <= 0)
+                    return;
+            }
+            else
+            {
+                // delete all revisions which age has passed
+                numberOfRevisionsToDelete = long.MaxValue;
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Revisions/RevisionsStorage.cs:495]: I have some doubts about that. If both of the settings `MinimumRevisionsToKeep` and `MinimumRevisionAgeToKeep` are `null` then we want to keep unlimited amount of revisions. What will happen now?

Comment (grisha-kotler) [reply src/Raven.Server/Documents/Revisions/RevisionsStorage.cs:495]: If both are null it won't reach this point.

Comment (ppekrol) [reply src/Raven.Server/Documents/Revisions/RevisionsStorage.cs:495]: Ok. Makes sense. I see the code now. The descriptions in the Studio match all our behaviors?

Comment (grisha-kotler) [reply src/Raven.Server/Documents/Revisions/RevisionsStorage.cs:495]: Yes

**Pull request merged**